src/diffusers/models/vae.py

```diff
@@ -1,12 +1,55 @@
+from dataclasses import dataclass
+
 import numpy as np
 import torch
 import torch.nn as nn
 
 from ..configuration_utils import ConfigMixin, register_to_config
 from ..modeling_utils import ModelMixin
+from ..utils import ModelOutput
 from .unet_blocks import UNetMidBlock2D, get_down_block, get_up_block
 
 
+@dataclass
+class DecoderOutput(ModelOutput):
+    """
+    Output of decoding method.
+
+    Args:
+        sample (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Decoded output sample of the model. Output of the last layer of the model.
+    """
+
+    sample: torch.FloatTensor = None
+
+
+@dataclass
+class VQEncoderOutput(ModelOutput):
+    """
+    Output of VQModel encoding method.
+
+    Args:
+        latents (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Encoded output sample of the model. Output of the last layer of the model.
+    """
+
+    latents: torch.FloatTensor = None
+
+
+@dataclass
+class AutoEncoderKLOutput(ModelOutput):
```

Comment:
Rename to `AutoencoderKLOutput` for consistency with the class (or alternatively rename `AutoencoderKL` to `AutoEncoderKL`. I prefer this change:

```suggestion
class AutoencoderKLOutput(ModelOutput):
```